Units/MMLAddon/plugins.pas

```diff
@@ -341,6 +345,23 @@ function TMPlugins.InitPlugin(Plugin: TLibHandle): boolean;
         StrDispose(PD);
       end;
     end;
+
+    Pointer(GetDelayedCode) := GetProcAddress(Plugin, PChar('GetDelayedCode'));
+    if (Assigned(GetDelayedCode)) then
+    begin
+      Pointer(GetDelayedCodeLength) := GetProcAddress(Plugin, PChar('GetDelayedCodeLength'));
+
+      if (Assigned(GetDelayedCodeLength)) then
+      begin
+       ArrC := GetDelayedCodeLength();
+       Writeln('Adding delayed code(', ArrC, ')');
+       PD := StrAlloc(ArrC);
+       PD := GetDelayedCode();
```

Comment:
This is a memory leak. You alloc'd a string then the plugin returns a new string.

You should rewrite GetDelayedCode to accept the PChar as a parameter.